inc/package.class.php

```diff
@@ -196,6 +197,7 @@ public function prepareInputForAdd($input) {
          if (rename($uploadedFile, $destination)) {
             $input['filesize'] = fileSize($destination);
             $input['dl_filename'] = basename($uploadedFile);
+            $input['name'] = $localFilename;
```

Comment:
Setting the name of the package is done while running the automatic action which parses it. Be aware the name will be overwritten one or two minute(s) later. Anyway, this is better to have something instead of a ID on display.

Reply:
but that name is never updated right now, which action do that?

Reply:
The name is not set when we upload a package. It is intended to be set later, when the package is parsed. Your idea to put the filename looks good for me; it sets a user friendly and temporary value until the parser finish its job. We just need to ensure the value is overwritten by the parser.

Reply:
Ok, this change is not only user friendly, it also fix a bug with "remove application" policy. When the name is empty you can't delete the policy.

Reply:
you're right. but putting the filename here may lead to an issue if you try to remove an application.

From the agent point of view, we need to send the application name which is similar to *com.something.app**. If t he agent received a filename it will not be able to identify the app to be removed.

Maybe we need to change a few things in the agent to handle the case where an app is not yet parsed or the parser fails.

@rafaelje What are the different ways you can use o identify an application in Android to remove it ?

Reply:
if the main name (com.example.app.vendor) is necesary for Android, I belive that We must change this logic to be compatible with other device's O.S. 

For that we must leave the auto-generated name and savethe correct device file in another colum, don't you think @btry?

Reply:
Not necessarly: we cannot deploy an apk on iOS, right ? Then we might format the MQTT message depending on the target OS of the package.

before considering changing this, let's ask to @hectorerb if iOS apps have a reversed domain name, and it can be used to detect if an app is installed.

Reply:
The only way that I can identify the app on Android is with the complete package name

Reply:
> Not necessarly: we cannot deploy an apk on iOS, right ? Then we might format the MQTT message depending on the target OS of the package.

Ok for iOS but what about Windows Phone or blackberry?